test/helpers.js

```diff
@@ -0,0 +1,57 @@
+import 'regenerator-runtime/runtime';
+import fs from 'fs';
+import path from 'path';
+import { tmpdir } from 'os';
+import { promisify } from 'util';
+import { config } from 'rxjs';
+
+const writeFile = promisify(fs.writeFile);
+
+const writeJSONFile = (filename, data) =>
+  writeFile(filename, JSON.stringify(data));
+
+/**
+ * There's some magic in order to create a relative path to the temporal directory. but mostly it does 3 things.
+ * 1) gets the dirname of the given path.
+ * 2) finds the relative from the temp file to the dirname found in step 1)
+ * 3) joins relativePath to the remaining fullPath (minus the dirnamePath)
+ */
+
+const getRelocatedPath = (dir, fullPath) => {
+  const dirnamePath = path.dirname(fullPath);
+  const relativePath = path.relative(dir, dirnamePath);
+  return path.join(path.join(relativePath, fullPath.split(dirnamePath)[1]));
+};
+
+export const temporaryConfigDirectory = async ({
```

Comment:
Have you considered using `fs.mkdtempSync`?

I believe it'll simplify a lot of this code. I'm not sure we get much benefit from using async calls here since we're in a test and `mkdtemp` is very quick.

Reply:
that's fine! I didn't think that it was a concern. 
I've changed everything to be synchronic